gammapy/cube/tests/test_fit.py

```diff
@@ -52,8 +52,10 @@ def background(geom):
 
 
 def edisp(geom, geom_etrue):
-    e_true = geom_etrue.get_axis_by_name("energy").edges
-    e_reco = geom.get_axis_by_name("energy").edges
+    e_true_axis = geom_etrue.get_axis_by_name("energy")
+    e_reco_axis = geom.get_axis_by_name("energy")
+    e_true = e_true_axis.edges * e_true_axis.unit
+    e_reco = e_reco_axis.edges * e_reco_axis.unit
```

Comment:
Why this is related to this PR?

Reply:
It was a bug that I just fixed along with this PR, because is appeared in the `npred` evaluation. The `edisp` was not initialised with units in the energy axis, which lead to incompatible map geometries (with incompatible units in the energy axis). So an easy addition of the background and spread map was not possible. Hope that makes sense...